api_app/analyzers_manager/observable_analyzers/bgp_ranking.py

```diff
@@ -0,0 +1,159 @@
+        pass
+
+    def run(self):
+        final_response = {}
+
+        # get ASN from ip
+
+        logger.info(f"Extracting ASN from IP: {self.observable_name}")
+        response = requests.get(
+            self.base_url + "/ipasn_history/?ip=" + self.observable_name,
+            timeout=self.timeout,
+        )
+        response.raise_for_status()
+        response = response.json()
+        asn = response.get("response", {}).popitem()[1].get("asn", None)
+        if not asn:
+            raise AnalyzerRunException(f"ASN not found in {response}")
+        logger.info(f"ASN {asn} extracted from {self.observable_name}")
+
+        # get ASN rank from extracted ASN
+        logger.info(f"Extracting ASN rank and position from ASN: {asn}")
+        response = requests.post(
+            self.base_url + "/json/asn",
+            data=json.dumps({"asn": asn}),
+            timeout=self.timeout,
+        )
+        response.raise_for_status()
+        response = response.json()
+        final_response["asn_description"] = response["response"].get(
+            "asn_description", None
+        )
+        final_response["asn_rank"] = response["response"]["ranking"].get("rank", None)
+        final_response["asn_position"] = response["response"]["ranking"].get(
+            "position", None
+        )
+        if final_response["asn_rank"] is None:
+            raise AnalyzerRunException(f"ASN rank not found in {response}")
+
+        logger.info(
+            f"""ASN rank: {final_response['asn_rank']},
+            position: {final_response['asn_position']},
+            from {self.observable_name}"""
+        )
+
+        if self.period:
+            # get ASN history from extracted ASN
+            logger.info(f"Extracting ASN history for period: {self.period}")
+            response = requests.post(
+                self.base_url + "/json/asn_history",
+                data=json.dumps({"asn": asn, "period": self.period}),
+                timeout=self.timeout,
+            )
+            response.raise_for_status()
+            response = response.json()
+            final_response["asn_history"] = response["response"].get(
+                "asn_history", None
+            )
+            if final_response["asn_history"] is None:
+                raise AnalyzerRunException(f"ASN history not found in {response}")
+            logger.info(
+                f"""ASN history: {final_response['asn_history']}
+                for {self.observable_name}"""
+            )
+        # we are using the ASN in a variable
+        # initially to avoid repetitive calculations
+        final_response["asn"] = asn
+
+        return final_response
+
+    def health_check(self) -> bool:
+        """
+        basic health check: if instance is up or not
+        """
+        try:
+            requests.head(self.base_url, timeout=self.timeout)
+        except requests.exceptions.RequestException:
+            return False
+
+        return True
```

Comment:
this is not required, the framework already does this once you set up a health check schedule

Reply:
My bad, overlooked a comment.